src/main/webapp/app/localci/build-queue/build-queue.component.ts

```diff
@@ -1,32 +1,76 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnDestroy, OnInit } from '@angular/core';
 import { ActivatedRoute } from '@angular/router';
-import { AccountService } from 'app/core/auth/account.service';
-import { BuildQueueService } from 'app/localci/build-queue/build-queue.service';
 import { BuildJob } from 'app/entities/build-job.model';
-import { faRefresh } from '@fortawesome/free-solid-svg-icons';
+import { JhiWebsocketService } from 'app/core/websocket/websocket.service';
+import { BuildQueueService } from 'app/localci/build-queue/build-queue.service';
 
 @Component({
     selector: 'jhi-build-queue',
     templateUrl: './build-queue.component.html',
     styleUrl: './build-queue.component.scss',
 })
-export class BuildQueueComponent implements OnInit {
+export class BuildQueueComponent implements OnInit, OnDestroy {
     queuedBuildJobs: BuildJob[];
     runningBuildJobs: BuildJob[];
-
-    //icons
-    faRefresh = faRefresh;
+    courseChannels: string[] = [];
 
     constructor(
         private route: ActivatedRoute,
-        private accountService: AccountService,
+        private websocketService: JhiWebsocketService,
         private buildQueueService: BuildQueueService,
     ) {}
 
     ngOnInit() {
         this.load();
```

Comment:
Couldn't the whole `load()` method be now streamlined, since also the initial load should be covered by the websocket?

Reply:
The problem is that updates are only sent over the WebSocket if build jobs are added or removed. This means if you open the Build Queue and no build jobs are added or removed at that moment, you will have to wait until you get your "first" data after having newly subscribed to the WebSocket topic. So, in the beginning, the table seems completely empty. That's why I left the load function, which is only called once when refreshing the website, so that the user has some initial data. The rest works over WebSockets.